### DISCUSSION TITLE
Request for more info on to get test working on macOS M1 - cannot connect to Docker daemon

### DISCUSSION THREAD

**Discussion by ringods** [kiegroup/act-js#34] [Q&A] [answered]
I'm as far as setting up my first test. I installed Act via Homebrew and set `ACT_BINARY` to bypass issues #23 and #33. Docker Desktop is installed and running for my MacBook Pro M1.

But the test run still fails to connect to the Docker daemon with this message:

```
thrown: "level=warning msg= ⚠ You are using Apple M1 chip and you have not specified container architecture, you might encounter issues while running act. If so, try running it with '--container-architecture linux/amd64'. ⚠ 

    time=\"2023-04-25T17:20:27+02:00\" level=error msg=\"failed to obtain container engine info: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?\"
    time=\"2023-04-25T17:20:27+02:00\" level=error msg=\"failed to obtain container engine info: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?\"
    Error: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
    "
```

What am I still missing regarding setup which prevents me from running a test with act-js?

**Comment by shubhbapna** [comment]
Ahh it might be because there is not option to pass `--container-architecture` to act-js. I will add a feature to pass this option

**Reply by shubhbapna** [reply, marked as answer]
@ringods Its been added in the latest release. I have also added an option to specify the docker daemon socket path in case it is not the default one "/var/run/docker.sock". Let me know if this works
